File: frontend/src/lib/thresholds.js
/**
 * Shared thresholds — single source of truth for the frontend.
 *
 * Values are injected at build time by Vite's `define` from shared/thresholds.json.
 * This module re-exports them as named constants for tree-shaking & IDE support.
 *
 * Usage:
 *   import { THRESHOLDS, classifyRisk, getScoreVariant } from '@/lib/thresholds'
 */

/* global __THRESHOLDS__ */
// eslint-disable-next-line no-undef
export const THRESHOLDS = __THRESHOLDS__

// ── Score thresholds ─────────────────────────────────────────────────────────
export const XAI_THRESHOLD = THRESHOLDS.score.xai          // 0.70
export const SAR_THRESHOLD = THRESHOLDS.score.sar          // 0.90

// ── Risk levels (sorted descending by min for fast classification) ───────────
const RISK_LEVELS = [...THRESHOLDS.risk_levels].sort((a, b) => b.min - a.min)

/**
 * Classify a numeric anomaly score into a risk level key.
 * @param {number} score - Anomaly score (0–1)
 * @returns {string} Risk level key: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW'
 */
export function classifyRisk(score) {
  const s = Number(score || 0)
  for (const level of RISK_LEVELS) {
    if (s >= level.min) return level.key
  }
  return 'LOW'
}

/**
 * Get the shadcn Badge variant for a given score.
 * @param {number} score - Anomaly score (0–1)
 * @returns {string} Badge variant: 'destructive' | 'warning' | 'outline'
 */
export function getScoreVariant(score) {
  const key = classifyRisk(score)
  return THRESHOLDS.ui.score_variant[key] || 'outline'
}

/**
 * Get the risk level metadata for a given score.
 * @param {number} score - Anomaly score (0–1)
 * @returns {{ key: string, label: string, color: string, min: number, max: number }}
 */
export function getRiskLevel(score) {
  const s = Number(score || 0)
  for (const level of RISK_LEVELS) {
    if (s >= level.min) return level
  }
  return RISK_LEVELS[RISK_LEVELS.length - 1]
}

// ── UI config re-exports ─────────────────────────────────────────────────────
export const HISTOGRAM_BUCKETS = THRESHOLDS.ui.histogram_buckets
export const FILTER_OPTIONS = THRESHOLDS.ui.filter_options
export const KPI_THRESHOLDS = THRESHOLDS.ui.kpi_variant
export const BUDGET_THRESHOLDS = THRESHOLDS.budget

File: shared/thresholds.py
"""
Shared thresholds — single source of truth.

This module loads shared/thresholds.json relative to the repository root
and exposes the values as plain Python objects. Used by:
  - backend/api  (config.py imports this)
  - backend/lambda_handler  (handler.py imports this)
  - backend/genai  (graph.py, flash_xai.py import this)
"""
import json
import os
from pathlib import Path

_THRESHOLDS_FILE = Path(__file__).parent / "thresholds.json"

if not _THRESHOLDS_FILE.exists():
    raise FileNotFoundError(f"Thresholds file not found: {_THRESHOLDS_FILE}")

with open(_THRESHOLDS_FILE, "r") as f:
    _data = json.load(f)

# ── Score thresholds ──────────────────────────────────────────────────────────
XAI_THRESHOLD: float = _data["score"]["xai"]            # 0.70
SAR_THRESHOLD: float = _data["score"]["sar"]            # 0.90
FLASH_RISK_ALTO: float = _data["score"]["flash_risk_alto"]  # 0.85

# ── Risk level bands (sorted descending by min) ──────────────────────────────
RISK_LEVELS: list[dict] = sorted(_data["risk_levels"], key=lambda r: r["min"], reverse=True)


def classify_risk(score: float) -> str:
    """Return the risk level key for a given anomaly score."""
    for level in RISK_LEVELS:
        if score >= level["min"]:
            return level["key"]
    return "LOW"


# ── Full config dict (for /api/config endpoint) ──────────────────────────────
THRESHOLDS_CONFIG: dict = _data
